casanovo/denovo/model.py

```diff
@@ -420,11 +426,24 @@ def on_predict_epoch_end(
                         calc_mz = self.peptide_mass_calculator.mass(
                             peptide_tokens, precursor_charge
                         )
-                        delta_mass_ppm = (
-                            (calc_mz - precursor_mz) / precursor_mz * 10**6
-                        )
-                        is_within_precursor_mz_tol = (
-                            abs(delta_mass_ppm) < self.precursor_mass_tol
+                        delta_mass_ppm = [
+                            (
+                                calc_mz
+                                - (
+                                    precursor_mz
+                                    - isotope * 1.00335 / precursor_charge
+                                )
+                            )
+                            / precursor_mz
+                            * 10**6
+                            for isotope in range(
+                                self.isotope_error_range[0],
+                                self.isotope_error_range[1] + 1,
+                            )
+                        ]
```

Comment:
**suggestion:** This is pretty difficult to read. What about adding a function:
```python
def calc_mass_error(calc_mz, obs_mz, isotope, charge):
    return (calc_mz - (obs_mz - isotope * 1.00335) / charge)) / obs_mz * 1e6
```

Then we could simplify this section to:
```python
delta_mass_ppm = [
    calc_mass_error(calc_mz, precursor_mz, precursor_charge, i) 
    for i in range(self.isotope_error_range[0], self.isotope_error_range[1] + 1)
]
```